### ISSUE TITLE
Help Wanted: Test new networking stack

### ISSUE THREAD

**Issue by joaomoreno** [Microsoft/vscode#22214]
If you were pinged to this thread, it's because you helped out in #18103.

With #21879, all network communication was moved to the Chrome's network stack. We'll try to fix the proxy issues over that layer, instead of having two layers.

The latest Insiders release is already out, please give it a try:

- [Windows](https://vscode-update.azurewebsites.net/latest/win32/insider)
- [macOS](https://vscode-update.azurewebsites.net/latest/darwin/insider)
- [Linux 64](https://vscode-update.azurewebsites.net/latest/linux-x64/insider)
- [Linux 32](https://vscode-update.azurewebsites.net/latest/linux-ia32/insider)

Please check whether you can:

1. Browse extensions
2. Install extensions

If stuff fails, please open the dev tools (<kbd>F1</kbd>, `Developer Tools`), try again and check the Network tab for failing requests.

**Note:** The `http.proxy` and `http.proxyStrictSSL` settings are ignored in this build, it doesn't matter what you put in there. The proxy settings are automatically picked up from your system's configuration. So, if stuff fails, the first thing you can check if whether your system is correctly configured to use your proxy.

---

@dfroze @akcorp2003 @fazouane-marouane @ywmail @abargstaedt @liaoya @R3V07T @hbuckle @ocean @joewood @martinmcwhorter @critesjm @dotalchemy @Jule- @diggamies @Efluminor @egamma @the-shank @raphabi @ununseptium @Platzer @williamchang @rfc791 @iamitava @ramdacxp @jrieken @abdinfos @ericc4 @lcrok @wiz @Harpagon @equinoxel @sattlt @nassoman @jithurjacob @sebastienraillet @rafaelcneves @FloMaetschke @JesusSerranoGL @kad @roblourens @leonibr @eolandezhang @xapienz @temurphy @atierant

**Comment by leonibr** [comment]
Windows link is dead

Em 08/03/2017 08:51, "João Moreno" <notifications@github.com> escreveu:

**Comment by rafaelcneves** [comment]
The "Linux 32" link is with "Linux 64" url, the same issue with "Linux 64" link

**Comment by fazouane-marouane** [comment]
The link to vscode insider
https://code.visualstudio.com/insiders

**Comment by xapienz** [comment]
Hello @joaomoreno, thank you for your effort.
I installed latest linux-64 deb from https://code.visualstudio.com/insiders (same build number as in tar.gz here - 1.11.0-1488956710_amd64).

After the update the problem #18920 didn't disappear, it still persists (`net::ERR_INSECURE_RESPONSE` in developer console). Moreover, after the update I cannot check for updates using "Help - Check for updates" (no messages appear in developer console).

Then I started code-insiders with `code-insiders --verbose` and noticed the message as in #17360 (`ERROR:cert_verify_proc_nss.cc(942)] CERT_PKIXVerifyCert`). So the workaround with certutil could help me (though nothing works without applying workaround).

In short, my problem seems to be not in proxy settings, it is the problem in accepting the corporate certificate despite using `http.proxyStrictSSL`.

**Comment by rafaelcneves** [comment]
I installed the latest version of code-insiders from the ppa, and I've tried with the links above too on linux x64.

And the error persists, I'm getting `net::ERR_CONNECTION_TIMED_OUT` in developer console. 

I have sure that proxy is correctly configured because I tried to simulate the same request on Postman and curl and there it works.

**Comment by equinoxel** [comment]
I've installed latest update for windows and I get the same error.

**Comment by FloMaetschke** [comment]
Hi @joaomoreno jaomoreno!

I still get the problem with ERR_TUNNEL_CONNECTION_FAILED.

I am using http_proxy setting in system env which works fine with npm.

**workbench.main.js:142 POST https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery net::ERR_TUNNEL_CONNECTION_FAILED(anonymous function)**

![image](https://cloud.githubusercontent.com/assets/7500948/23705366/cc6971d0-0409-11e7-8782-35603da0bd75.png)

```
workbench.main.js:142 POST https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery net::ERR_TUNNEL_CONNECTION_FAILED(anonymous function) @ workbench.main.js:142n.Class.derive._oncancel @ workbench.main.js:32t.xhrRequest @ workbench.main.js:142e.request @ workbench.main.js:142n.request @ workbench.main.js:142(anonymous function) @ workbench.main.js:142then @ workbench.main.js:32e.queryGallery @ workbench.main.js:142e.query @ workbench.main.js:142e.queryGallery @ workbench.main.js:97t.query @ workbench.main.js:138t.doSearch @ workbench.main.js:138(anonymous function) @ workbench.main.js:138e.queue @ workbench.main.js:32(anonymous function) @ workbench.main.js:32(anonymous function) @ workbench.main.js:32g @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_completed @ workbench.main.js:32(anonymous function) @ workbench.main.js:32
4workbench.main.js:103 An unknown error occurred. Please consult the log for more details.e.doShow @ workbench.main.js:103e.show @ workbench.main.js:103(anonymous function) @ workbench.main.js:103e.show @ workbench.main.js:103t.onError @ workbench.main.js:138(anonymous function) @ workbench.main.js:138done @ workbench.main.js:32v @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_chainedError @ workbench.main.js:32n @ workbench.main.js:32v @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_error @ workbench.main.js:32(anonymous function) @ workbench.main.js:32done @ workbench.main.js:32v @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_error @ workbench.main.js:32(anonymous function) @ workbench.main.js:32done @ workbench.main.js:32v @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_error @ workbench.main.js:32u @ workbench.main.js:32(anonymous function) @ workbench.main.js:32v @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_chainedError @ workbench.main.js:32n @ workbench.main.js:32then @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_setCompleteValue @ workbench.main.js:32v @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_chainedError @ workbench.main.js:32n @ workbench.main.js:32v @ workbench.main.js:32enter @ workbench.main.js:32_run @ workbench.main.js:32_error @ workbench.main.js:32
```

**Comment by JesusSerranoGL** [comment]
Insiders version worked for me! Thanks!

**Comment by joaomoreno** [comment]
Fixed the links, sorry guys. 😨

**Comment by joaomoreno** [comment]
@xapienz I'm now investigating on how to support the scenario of untrusted certificates. 👍 

@rafaelcneves Strange. Chrome works well in your system? Do you have Firefox? If so, do you manually configure the proxy settings in it?

@FloMaetschke I mean the _Windows Proxy Settings(IE)_. The `HTTP_PROXY` doesn't work anymore.

**Comment by FloMaetschke** [comment]
@joaomoreno Proxy Settings are set but not working.

**Comment by joaomoreno** [comment]
@FloMaetschke Does Google Chrome work in your machine? How about Firefox?

**Comment by joaomoreno** [comment]
I'll try to be active in our Gitter room: just ping me there directly. [![Gitter](https://img.shields.io/badge/chat-on%20gitter-blue.svg)](https://gitter.im/Microsoft/vscode)

**Comment by martinmcwhorter** [comment]
Windows

I am still seeing the same non working result as @FloMaetschke is:

    (failed) net::ERR_TUNNEL_CONNECTION_FAILED

while requesting https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery

**Comment by rafaelcneves** [comment]
@joaomoreno Yes, but I need to configure it manually as well as Firefox.

I'm using Ubuntu, and it doesn't have an effective system proxy configuration with auth options, because of this I need to set HTTP_PROXY env vars on profile, and it works at most of applications.

**Comment by joaomoreno** [comment]
@rafaelcneves Can you show me the Chrome network configuration dialog?

**Comment by joaomoreno** [comment]
@Thiruppathi Yes, you can do it, I just don't expose it yet.

[Can you give this a try?](https://www.chromium.org/developers/design-documents/network-settings)

So, that' be:

```
code --proxy-server=socks5://myproxyaddress
```

Let me know how it goes.

**Comment by Thiruppathi** [comment]
@joaomoreno Thanks for the quick response.
I've tried the following command

`code --proxy-server=socks5://localhost:8888`

and got the following response

`open: unrecognized option --proxy-server=socks5://localhost:8888'`

I've tried the same with VS Code Insiders as well using alias(code-insiders)

`code-insiders --proxy-server=socks5://localhost:8888`

I'm not sure how system(macOS) level configuration works; didn't want to mess it up;  😟

**Comment by rafaelcneves** [comment]
@joaomoreno I set it on chrome passing `--proxy-server` option, on linux chrome does not have support to proxy, that I can only get this by using a plugin.

I have that system settings dialog, but I can't set auth on that.

**Comment by joaomoreno** [comment]
@Thiruppathi You still have a very old Code launcher set up. Open your `bashrc` and remove the code alias there. Then, in Code go <kbd>F1</kbd> and run the `Install 'code' command in PATH` command.

**Comment by martinmcwhorter** [comment]
Is there a command line switch to turn off proxy strict ssl?

**Comment by joaomoreno** [comment]
@rafaelcneves Can you try passing the very same `--proxy-server` opt to Code?

To confirm: you can't set the systems dialog due to lack of permissions?

**Comment by joaomoreno** [comment]
@martinmcwhorter Not yet.

**Comment by Thiruppathi** [comment]
@joaomoreno Thanks for the instruction; now I'm able to launch code with 

`code --proxy-server=socks5://localhost:8888`

Unfortunately, I'm unable to access the url I wanted to access from code.
Got the following error.

`443: Operation timed out`

**Comment by joaomoreno** [comment]
Can you show me the error in the devtools? A screenshot?

**Comment by rafaelcneves** [comment]
@joaomoreno Not, I'm talking about proxy user settings, my proxy config is like this: `user:password@proxy:port`

With this `--proxy-server` option I get `net::ERR_TUNNEL_CONNECTION_FAILED` probably because this proxy user is not set.

**Comment by joaomoreno** [comment]
@rafaelcneves I don't understand. I thought you had a working value for `--proxy-server` which you use for Chrome:

> @joaomoreno I set it on chrome passing --proxy-server option

If you pass the same value to Code, you get that `net::ERR_TUNNEL_CONNECTION_FAILED` error?

**Comment by rafaelcneves** [comment]
@joaomoreno It's because chrome throws me a dialog to pass user and password, code don't do this.

**Comment by rafaelcneves** [comment]
This is what happen on chrome:

![proxy](https://cloud.githubusercontent.com/assets/421754/23715600/01c2dfc2-040c-11e7-9a99-6dd1fbf19093.png)

**Comment by Jule-** [comment]
@joaomoreno Indeed I got the same issue than @rafaelcneves. I am behind an **authenticated** proxy and in Windows proxy settings you can't save your credentials (and you don't want to, even if most of the time you have no choice for dev purpose that saving your password in clear in your `http_proxy` environment variable...).
So we need some alternative to the proxy credential popup shown in the @joaomoreno screenshot.

And I don't know if it matters but my proxy is not a socks proxy.

**Comment by lcrok** [comment]
@joaomoreno Installed this version and everything is working perfectly. I could remove the http_proxy setting and it got the system wide proxy. No more errors. Thanks for working on that.

**Comment by FloMaetschke** [comment]
I totally agree with @Jule- !

But anyway, why don't you just add these 3 lines of code for checking HTTP_PROXY and setting their values to the new single network stack on startup? HTTP_PROXY is a common way, why don't you support it anymore?

**Comment by joaomoreno** [comment]
@rafaelcneves @Jule- Very interesting. I will try to reproduce your setup. I assume the ideal scenario is that Code shows you a similar auth popup and just remembers the credentials from then on?

Also, does it work if you inline the credentials in the proxy url? `user:pass@host

@lcrok Great to know thanks. What is your OS btw?

**Comment by FloMaetschke** [comment]
@joaomoreno Yes! I am sorry i can't be in gitter because its completely blocked by corporate proxy.

I tried adding Fiddler 4. Configured my proxy there as gateway, so it forwards the request.

The result is, that the proxy responds with a _HTTP/1.1 407 Proxy Authentication Required_.

Now i used chrome with the option to automatically detect settings (system proxy config) and it uses fiddler4, too.
Chrome then shows up a proxy credentials input like @rafaelcneves already mentioned above.

**Comment by lcrok** [comment]
> @lcrok Great to know thanks. What is your OS btw?

@joaomoreno I'm using Windows 10 anniversary edition. Behind a corporate proxy authenticated by smart cards. I don't have any username and password to put in, but it is now working with the authentication handled by windows. It is working just like the browser does. [Original Issue Post](https://github.com/Microsoft/vscode/issues/18103#issuecomment-275683253).

**Comment by Jule-** [comment]
@joaomoreno 
> I assume the ideal scenario is that Code shows you a similar auth popup and just remembers the credentials from then on?

Yes! 😃 

> Also, does it work if you inline the credentials in the proxy url? `user:pass@host:port`

You can't do that in Windows proxy settings and you told us that the `http.proxy` parameter in Code is not used anymore, so no it does not work.

**Comment by joaomoreno** [comment]
@Jule- Alright, let me see if I can sketch up that auth dialog.

**Comment by akcorp2003** [comment]
I'm using MacOS El Capitan 10.11.6. I have the proxy configured for my system. I didn't change anything in the preferences for VS Code.

However, I still can't install extensions. 

<img width="806" alt="screen shot 2017-03-09 at 8 56 27 am" src="https://cloud.githubusercontent.com/assets/12402936/23758240/5855cd96-04a6-11e7-80e3-9d3ce35cba63.png">

<img width="803" alt="screen shot 2017-03-09 at 8 57 43 am" src="https://cloud.githubusercontent.com/assets/12402936/23758271/7f4cd778-04a6-11e7-9244-32e6d886744e.png">

Here are my network settings. Not showing the addresses for obvious reasons. Also, I have a PAC file under Automatic Proxy Configuration.

<img width="654" alt="screen shot 2017-03-09 at 8 59 55 am" src="https://cloud.githubusercontent.com/assets/12402936/23758399/e127eae6-04a6-11e7-88cf-a7b622d1fb30.png">

Please let me know if I can provide anything else. Thanks for the work you have done fixing this issue!! :D

**Comment by joaomoreno** [comment]
Leaving this here for my own reference.

I was able to install and configure Squid proxy on my dev machine. The magic sauce is:

```
auth_param basic program /usr/local/opt/squid/libexec/basic_ncsa_auth /Users/joao/passwords
auth_param basic realm proxy
acl authenticated proxy_auth REQUIRED
http_access allow authenticated
```

**Comment by Jule-** [comment]
@joaomoreno no the credentials are saved. In fact when you validate your credentials you have the Chrome popup near the address bar on the key icon that asking you if you want to save them. I'm not sure about that but I think it can be synced with your Google account.

**Comment by joaomoreno** [comment]
Maybe they go onto the OS's keychain?

**Comment by Harpagon** [comment]
Error still persist.Ubuntu 14.04 x32 with corporate proxy (squid/3.2.3) with auth.
`http_proxy` and `https_proxy` environment variables exists.
Google Chrome not installed.

`--proxy-server="http(s)://user:pw@ip:port/"`
raising : `net::ERR_NO_SUPPORTED_PROXIES`

![default](https://cloud.githubusercontent.com/assets/6270201/23784587/ac202ad2-056b-11e7-8e1e-1e3a2f9f0b2b.png)

**Comment by fazouane-marouane** [comment]
Have you tried passing only "--proxy-auto-detect" ?

**Comment by FloMaetschke** [comment]
> Maybe they go onto the OS's keychain?

@joaomoreno 
They are saved into chromes credential storage. (settings-> search for password)

**Comment by joaomoreno** [comment]
Auth support coming: #22369

**Comment by wiz** [comment]
Works with `--proxy-server=http://127.0.0.1:3128` but does not catch system proxy settings and/or `http_proxy` env var.

**Comment by xuthief** [comment]
golint:
Error: Command failed: /usr/local/bin/go get -u -v github.com/golang/lint/golint
github.com/golang/lint (download)
Fetching https://golang.org/x/tools/go/gcexportdata?go-get=1
https fetch failed: Get https://golang.org/x/tools/go/gcexportdata?go-get=1: dial tcp 216.239.37.1:443: i/o timeout

**Comment by xuthief** [comment]
start with `code --proxy-server=socks5://localhost:1080`  
but failed with tcp timeout:
```
golint:
Error: Command failed: /usr/local/bin/go get -u -v github.com/golang/lint/golint
github.com/golang/lint (download)
Fetching https://golang.org/x/tools/go/gcexportdata?go-get=1
https fetch failed: Get https://golang.org/x/tools/go/gcexportdata?go-get=1: dial tcp 216.239.37.1:443: i/o timeout
```

**Comment by joaomoreno** [comment]
@wiz @FloMaetschke We can add some sprinkles on top of everything 🌸  to detect that `HTTP_PROXY` env var.

@wiz But in your case, if you have configured your system... it should have picked up the settings. Which OS are you on?

**Comment by wiz** [comment]
Ubuntu 16.04. I have a global proxy set up and a chrome policy in `/etc/opt/chrome/policies/managed/proxy.json` to use same localhost/cntlm proxy.

**Comment by joaomoreno** [comment]
@wiz Like here?

![image](https://cloud.githubusercontent.com/assets/22350/23710325/9fbfd9c2-041c-11e7-9cb7-ae05968e96ce.png)

**Comment by wiz** [comment]
@joaomoreno yeah. with http and socks proxies set.

**Comment by wiz** [comment]
Hmm.. I've added the socks proxy and `code-insiders` now appear to work without command line proxy argument.

**Comment by joaomoreno** [comment]
@wiz 🤘

**Comment by FloMaetschke** [comment]
@joaomoreno is your vscode version with username / password dialog already an insiders-build ?

**Comment by joaomoreno** [comment]
Nah man, it'll take a while. Busy busy.

**Comment by ocean** [comment]
I'm on Mac and have an authenticated corporate proxy with an internal self-signed certificate.

I've tried the latest Insiders build (today's version) and I cannot connect to the marketplace to browse extensions, install extensions or check for program updates.

I have tried with the `--proxy-server` command line parameter set, and without.

I get exactly the same `net::ERR_TUNNEL_CONNECTION_FAILED` error as [FloMaetschke does here](https://github.com/Microsoft/vscode/issues/22214#issuecomment-285037515).

I assume this is because my authenticated proxy requires #22369 😃 

Thanks for the progress @joaomoreno, keep going! 👍

**Comment by SQLDBAWithABeard** [comment]
Hi,

Invited here by Sean McBreen as was having the issues installing extensions as mentioned here.

The insider build allowed me to install extensions but after installing the mssql extension. I got this error

Initializing SQL tools service for the mssql extension.
Note: mssql commands will be available after installing the service.

Platform: win32, x86_64 (Windows)

Installing SQL tools service to C:\Users\XXXXXXXXXXX\.vscode-insiders\extensions\ms-mssql.mssql-0.3.0\sqltoolsservice\0.3.0.5\Windows.
Downloading https://download.microsoft.com/download/A/A/2/AA22B2D6-94D8-4C82-A388-D0246F567BBF/microsoft.sqltools.servicelayer-win-x64-netcoreapp1.0.zip
[ERROR] Error: Request error: ETIMEDOUT

As I had already gone through the process of manually downloading and installing I was able to copy the files and all was well

**Comment by FloMaetschke** [comment]
any update so far?

**Comment by diev** [comment]
This version (for Windows) has solved my #23166 after I had seen in Developer Tools many undocumented requests to ```https://*.vsassets.io/*``` (our corporate proxy has white lists of allowed domain urls for every job - **please write your VS Code url requirements somewhere for attention of proxy admins and CISO!**).

Yes, I can:
1. Browse extensions
2. Install extensions

with common IE proxy settings set by GPO, where same IP:3128 for every protocol.

**Comment by jgrumboe** [comment]
Hi, tried insider version 1.11.0 from today and i have to report the following:
1. setting proxy as environment and/or settings.json didn't work - still direct TCP connections to internet hanging in SYN_SENT
2. starting code-insider with "--proxy-server=http://proxy.server:8080" did help! woohoo!
I can browse/install extensions and check for updates.

Strangly when i check with "netstat -tnp|grep code" for TCP connections there are still some SYN_SENT to 40.77.226.250:443.

Can i be any more help for a final solution?
Regards,
Johannes

**Comment by jgrumboe** [comment]
Just tried it with official code 1.10.2 and "--proxy-server" and extensions are now also working for this build under RHEL7!

**Comment by ocean** [comment]
Hi @joaomoreno I see your PR for this feature has passed the tests (4 days ago), does that it mean will in the Insiders build soon for us to test?

**Comment by diev** [comment]
Somedays our admins had problems to pass the Windows Updates through our corporate proxy, after our WSUS got died. These problems were solved with
```
netsh.exe winhttp import proxy source=ie
```
May it be a better way than to write the settings for every application?

**Comment by danielbarla** [comment]
I'm using release build 1.11.2, and I'm getting an issue that's essentially explained by the note above:

    Note: The http.proxy and http.proxyStrictSSL settings are ignored in this build, it doesn't matter what you put in there. The proxy settings are automatically picked up from your system's configuration.

I was just about to log a separate issue for this, when I ran across this topic.

Is `http.proxy` supposed to be ignored in the current release version?  It certainly seems to be the case now, and I'm pretty sure it wasn't happening in previous builds.  Can confirm that VS Code is following Chrome's implementation, i.e. changing the system proxy instantly takes effect in VS Code (we're running Cntlm in `-f -v` mode to observe it being hit - we haven't been able to make `http.proxy` have any effect lately).

**Comment by FloMaetschke** [comment]
- VSCode Version: Code - Insiders 1.12.0-insider (740f6620234326925592c40d2ec858bd457f9f6c, 2017-04-20T06:11:56.973Z)
- OS Version: Windows_NT ia32 6.1.7601
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|angular1-code-snippets|alexandersage|0.0.3|
|githistory|donjayamanne|0.2.0|
|KillTasksButton|dtrajano|0.0.4|
|output-colorizer|IBM|0.1.0|
|latex-workshop|James-Yu|2.1.4|
|color-highlight|naumovs|2.1.2|
|material-icon-theme|PKief|2.0.4|
|Material-theme|zhuangtongfa|2.4.17|;


---

Steps to Reproduce:

1. Start vscode without cli parameters and open extensions panel.
2. Search for new extenions.

![image](https://cloud.githubusercontent.com/assets/7500948/25222173/01cb4fc2-25b8-11e7-9397-aa1bcb6f0412.png)

Search not working, no results. Request fails.



***I was able to get it working with fiddler4.***

**Steps to Reproduce:**

1. Start vscode with cli parameter:  --ignore-certificate-errors

2. Open extensions panel and search for new extenions.


How long until i can enter proxy credentials in a form from vscode on first connect?
Please include an option to ignore certificate and to save this settings.

**Comment by odulkanberoglu** [comment]
After installing version 1.11.2, extension downloads are function correctly. However after installing C/C++ extension and restarting VS Code, following output is observed on Output window.

```
Updating C/C++ dependencies...

Downloading package 'C/C++ language components (Windows)' Failed at stage: downloadPackages
failed (error code '401')

Finished
```

**Comment by opticyclic** [comment]
I can confirm that using 1.12.0 the issue reported here https://github.com/Microsoft/vscode/issues/7140  has now been resolved and I can browse and install extensions.

Windows with a corporate firewall configured with a pac.

**Comment by martinmcwhorter** [comment]
I can confirm I am still having this issue trying to access the marketplace through a corporate proxy

VSCode: 1.12.1
OS: Windows 7
Proxy configured through PAC.

When I try to access @recommended extensions I get this in the debug console:

![image](https://cloud.githubusercontent.com/assets/1194490/26580226/2069b270-452f-11e7-81b7-7bf97ec73a77.png)

I have deleted all of my proxy config settings from VSCode, exited and restarted VSCode I get the same result. 

If I only delete the first line, and leave in the second line of the config:

    //"http.proxy": "http://<user>:<pass>@<proxyhost>:<port>",
    "http.proxyStrictSSL": false,

It still does not work. I get the same result.

**Comment by akcorp2003** [comment]
Currently on Mac OSX, VS Code 1.12.2, I am able to search for extensions now, however, the thumbnails do not appear and if I click on "install," I get an error message. 


<img width="1440" alt="screen shot 2017-05-31 at 2 09 05 pm" src="https://cloud.githubusercontent.com/assets/12402936/26654229/e1eba4ea-460a-11e7-8c16-7901b52c2303.png">

**Comment by ywmail** [comment]
Please add `no.proxy` setting support. Some extensions honour vscode's `http.proxy` setting, but some network requests from these extensions like testing local APIs, should not go through proxy.

I'm using this extension: REST Client `humao.rest-client`
I can send request to github.com, like:
```
GET https://github.com/ HTTP/1.1
content-type: application/json
```

But I can't send request to local APIs when `http.proxy` is set. 
```
GET http://127.0.0.1:18081/api/test / HTTP/1.1
content-type: application/json
```

remove the `http.proxy` setting is OK to send the request.

**Comment by HolyGeneralK** [comment]
Seconded on the `no.proxy` or `http.noproxy` support (or at least make sure that we can use a system setting). If you'd like me to create a new issue and link it here I can. I would be able to do some limited testing if needed.

**Comment by joaomoreno** [comment]
Thanks for all the help people!
